app/src/lib/stores/helpers/ahead-behind-updater.ts

```diff
@@ -77,15 +56,15 @@ export class AheadBehindUpdater {
       }
 
       this.executeTask(from, to, (error, result) =>
-        error !== null ? reject(error) : resolve(result)
+        error !== null ? reject(error) : resolve(result || null)
       )
     })
   }
 
   private executeTask = (
     from: string,
     to: string,
-    callback: (error: Error | null, result: IAheadBehind | null) => void
+    callback?: (error?: Error, result?: IAheadBehind) => void
```

Comment:
The TS definitions of the `queue` package are quite low quality, since they have a few discrepancies with the actual logic (plus they don't use generics so the result is typed as `any`). These are some of the discrepancies:

- The actual `callback` function is always passed to each job ([source](https://github.com/jessetane/queue/blob/5.0.1/index.js#L143)) but the typings say that it can be `undefined`.
- The type of the `error` argument is `Error | undefined`, when the convention in Node.js is to use `null` for error arguments in callbacks ([source](https://nodejs.org/en/knowledge/errors/what-are-the-error-conventions/)). It also [internally uses `null`](https://github.com/jessetane/queue/blob/5.0.1/index.js#L146) when it throws an error itself (this doesn't matter much because the library does a [loose null comparison](https://github.com/jessetane/queue/blob/5.0.1/index.js#L106) for the error object).
- The `result` type is `Object | undefined`, while the `undefined` part makes sense for when we only throw an error, forcing a `Object` type prevents us from returning `null` (since `null` is not a subtype of `Object`).

I'm considering fixing their types and sending a PR to that project, but for now I've decided to make very small and low-risk changes to this logic (return `undefined` instead of `null`) to workaround the type issues.

Reply:
Edit: I've ended up creating 1020e07c08176ed2064d3442ab062c0d57d31749 to change a bit the logic and move the jobs to return a `Promise`. With this I can bypass some of the issues on the types of `queue` 😅